pkg/aws/actuator/actuator_test.go

```diff
@@ -271,6 +275,98 @@ func TestGenerateUserName(t *testing.T) {
 	}
 }
 
+func TestUpgradeable(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+
+	codec, err := minterv1.NewCodec()
+	if err != nil {
+		t.Fatalf("failed to set up codec for tests: %v", err)
+	}
+
+	tests := []struct {
+		name                    string
+		mode                    v1.CloudCredentialsMode
+		existing                []runtime.Object
+		expectedStatus          configv1.ConditionStatus
+		expectedReason          string
+		overrideUpcomingSecrets []types.NamespacedName
+	}{
+		{
+			name: "mint mode with root cred",
+			mode: v1.CloudCredentialsModeMint,
+			existing: []runtime.Object{
+				testRootSecret(),
+			},
+			expectedStatus: configv1.ConditionTrue,
+		},
+		{
+			name: "implicit mint mode with root cred",
+			mode: v1.CloudCredentialsModeDefault,
+			existing: []runtime.Object{
+				testRootSecret(),
+			},
+			expectedStatus: configv1.ConditionTrue,
+		},
+		{
+			name:           "mint mode with missing root cred",
+			mode:           v1.CloudCredentialsModeMint,
+			existing:       []runtime.Object{},
+			expectedStatus: configv1.ConditionFalse,
+			expectedReason: constants.MissingRootCredentialUpgradeableReason,
+		},
+		{
+			name:           "implicit mint mode with missing root cred",
+			mode:           v1.CloudCredentialsModeDefault,
+			existing:       []runtime.Object{},
+			expectedStatus: configv1.ConditionFalse,
+			expectedReason: constants.MissingRootCredentialUpgradeableReason,
+		},
+		{
+			name: "manual mode with 4.6 secrets pre-provisioned",
```

Comment:
maybe update the name to something like `manual mode with future version secrets pre-provisioned`